### ISSUE TITLE
type of json column in auto-created table

### ISSUE THREAD

**Issue by dmossakowski** [an0r0c/kafka-connect-transform-tojsonstring#3]
Thanks for a great transform. I'm trying to use it and it creates a new table for the topic but the jsondata field is set to type 'text' instead of json in PostgreSQL. Is that correct or am I doing something wrong?

This is the config:
connector.class=io.confluent.connect.jdbc.JdbcSinkConnector
auto.create=true
..
transforms=tojson
transforms.tojson.type=com.github.cedelsb.kafka.connect.smt.Record2JsonStringConverter$Value
transforms.tojson.json.string.field.name=jsondata
transforms.tojson.post.processing.to.xml=false

This is the SQL in pgAdmin for the newly created table:

CREATE TABLE public."TILLCODES_1"
(
    jsondata text COLLATE pg_catalog."default" NOT NULL,
    "ID" text COLLATE pg_catalog."default" NOT NULL,
    CONSTRAINT "TILLCODES_1_pkey" PRIMARY KEY ("ID")
)

**Comment by an0r0c** [comment]
Unfortunately the JDBC Sink generates Text columns and the SMT cannot really influence that. That mapping from Connect data type to postgres datatype is deep in the java code of connect itself. The only way I found to influence that would had been to directly modify code in kafka connect jdbc sink. 

If you google about that you'll see many folks having issues with the mapping between kafka connect record data types and DB datatypes as it is quite static in JDBC Sink Connector. 

How I workarrounded it in several use-cases in the past was to: 
1. Disable the auto creation of the table 
2. Create the table manually (or by the Schema Management of the target system) with proper data types but same table and field names. 

On some pipes with oracle as target this worked with XML columns. I never tried it with postgres but I assume it should work
